File: manual/code/generate_strategies/IR(summary)/java/java/function_request_head.java
package com.example;

import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;

import java.io.IOException;
import java.util.concurrent.TimeUnit;

public class RequestHandler {
    public String handle_request(String url) {
        class RequestHandler {
            public String handle_request(String requestUrl) {
                OkHttpClient client = new OkHttpClient.Builder()
                        .connectTimeout(5, TimeUnit.SECONDS)
                        .followRedirects(false)
                        .build();

                Request request = new Request.Builder()
                        .url(requestUrl)
                        .head()
                        .build();

                try {
                    Response response = client.newCall(request).execute();
                    int statusCode = response.code();
                    response.close();

                    if (statusCode == 200) {
                        return "success";
                    } else if (statusCode >= 300 && statusCode < 400) {
                        return "redirect " + statusCode;
                    } else {
                        return "error " + statusCode;
                    }
                } catch (IOException e) {
                    return "timeout";
                }
            }
        }
        return new RequestHandler().handle_request(url);
    }
}

File: manual/data/python/src/function_request_head.py
import requests

class RequestHandler:
    def handle_request(self, url: str) -> str:
        try:
            response = requests.head(url, timeout=5, allow_redirects=False)
        except requests.exceptions.Timeout:
            return "timeout"

        if response.status_code == 200:
            return "success"
        elif 300 <= response.status_code < 400:
            return f"redirect {response.status_code}"
        else:
            return f"error {response.status_code}"
